pages/agent/v3/hooks.md

```diff
@@ -137,29 +136,34 @@ they are run as part of each job:
 
 ### Creating job lifecycle hooks
 
-Job lifecycle hooks are sourced for every job an agent accepts.
-Use job lifecycle hooks to prepare for jobs, override default behavior, or clean up after jobs that have finished.
-For example, use the `environment` hook to set a job's environment variables or the `pre-exit` hook to delete temporary files and remove containers.
-If your hook is related to the startup or shutdown of the agent, consider [agent lifecycle hooks](#agent-lifecycle-hooks) for those tasks instead.
+Job lifecycle hooks are sourced for every job an agent accepts. Use job lifecycle hooks to prepare for jobs, override default behavior, or clean up after jobs that have finished. For example, use the `environment` hook to set a job's environment variables or the `pre-exit` hook to delete temporary files and remove containers. If your hook is related to the startup or shutdown of the agent, consider [agent lifecycle hooks](#agent-lifecycle-hooks) for those tasks instead.
 
 Job lifecycle hooks have access to all the standard [Buildkite environment variables](/docs/pipelines/environment-variables).
 
-Job lifecycle hooks are copied to `$TMPDIR` directory and *sourced* by the agent's default shell.
-This has a few implications:
+Job lifecycle hooks are copied to `$TMPDIR` directory and *sourced* by the agent's default shell. This has a few implications:
+
+#### \<aside\>
+#### 📝 A note on sourcing for shellscripting nerds
+
+We use the word "sourcing" in this article, but it's not strictly speaking true; we're lying by omission a little bit. Instead, the agent uses a process called ["the scriptwrapper"](https://github.com/buildkite/agent/blob/1a5f05029cc363a984188c441f938dd316dedd16/hook/scriptwrapper.go).
+
+This process notes down the environment variables prior to a hook run, sources that hook, and then compares the environment variables after the hook run to the environment variables before the hook run.
+
+Any environment variables that were added, changed or removed are then exported to the subsequent phases and hooks. Functionally, this is very similar to how `source` would work, but it's not quite the same, and it means that if you're relying on some very specific pieces of shellscripting arcana, you might find that things don't work quite as you expect.
+
+The reason that we do this is that there's no shared bash environment between two different hooks on the same job; functionally, each hook is run in its own shell, which are orchestrated through the Agent's Go code. This means that if you set an environment variable in one hook, it wouldn't be available in the next hook without this scriptwrapper process.
+
+#### \<\/aside\>
```

Comment:
![CleanShot 2023-06-15 at 16 56 07](https://github.com/buildkite/docs/assets/11305534/4b51047e-43d2-4a57-bf9c-b859dd5dacc4)

I think you've got one extra escape backslash

Reply:
ah dangit, i rewrote this section an put it in a details block, but now it's back... will sort it out